packages/transformers/src/tests/idiomatic-test.ts

```diff
@@ -0,0 +1,392 @@
+    }
+  }
+  `;
+  checkTransform(input, expected);
+});
+
+test('@queryAssignedNodes (default slot)', () => {
+  const input = `
+  import {LitElement} from 'lit';
+  import {queryAssignedNodes} from 'lit/decorators.js';
+  class MyElement extends LitElement {
+    @queryAssignedNodes()
+    listItems: NodeListOf<HTMLElement>;
+  }
+  `;
+
+  const expected = `
+  import {LitElement} from 'lit';
+  class MyElement extends LitElement {
+    get listItems() {
+      return this.renderRoot
+        ?.querySelector('slot:not([name])')
+        ?.assignedNodes();
+    }
+  }
+  `;
+  checkTransform(input, expected);
+});
+
+test('@queryAssignedNodes (with slot name)', () => {
+  const input = `
+  import {LitElement} from 'lit';
+  import {queryAssignedNodes} from 'lit/decorators.js';
+  class MyElement extends LitElement {
+    @queryAssignedNodes('list')
+    listItems: NodeListOf<HTMLElement>;
+  }
+  `;
+
+  const expected = `
+  import {LitElement} from 'lit';
+  class MyElement extends LitElement {
+    get listItems() {
+      return this.renderRoot
+        ?.querySelector('slot[name=list]')
+        ?.assignedNodes();
+    }
+  }
+  `;
+  checkTransform(input, expected);
+});
+
+test('@queryAssignedNodes (with flatten)', () => {
+  const input = `
+  import {LitElement} from 'lit';
+  import {queryAssignedNodes} from 'lit/decorators.js';
+  class MyElement extends LitElement {
+    @queryAssignedNodes('list', true)
+    listItems: NodeListOf<HTMLElement>;
+  }
+  `;
+
+  const expected = `
+  import {LitElement} from 'lit';
+  class MyElement extends LitElement {
+    get listItems() {
+      return this.renderRoot
+        ?.querySelector('slot[name=list]')
+        ?.assignedNodes({flatten: true});
+    }
+  }
+  `;
+  checkTransform(input, expected);
+});
+
+test('@queryAssignedNodes (with selector)', () => {
+  const input = `
+  import {LitElement} from 'lit';
+  import {queryAssignedNodes} from 'lit/decorators.js';
+  class MyElement extends LitElement {
+    @queryAssignedNodes('list', false, '.item')
+    listItems: NodeListOf<HTMLElement>;
+  }
+  `;
+
+  const expected = `
+  import {LitElement} from 'lit';
+  class MyElement extends LitElement {
+    get listItems() {
+      return this.renderRoot
+        ?.querySelector('slot[name=list]')
+        ?.assignedNodes()
+        ?.filter((node) =>
+          node.nodeType === Node.ELEMENT_NODE &&
+            node.matches('.item')
+        );
+    }
+  }
+  `;
+  checkTransform(input, expected);
+});
+
+test('@eventOptions', () => {
+  const input = `
+  import {LitElement} from 'lit';
+  import {eventOptions} from 'lit/decorators.js';
+  class MyElement extends LitElement {
+    @eventOptions({once: true})
+    _onClick(event) {
+      console.log('click', event.target);
+    }
+  }
+  `;
+
+  const expected = `
+  import {LitElement} from 'lit';
+  class MyElement extends LitElement {
+    constructor() {
+      super(...arguments);
+      this._onClick = {
```

Comment:
Hm, I don't think this is a safe change. Before _onClick was a function, now it's an object.

Reply:
Hmm true, this works well for a Lit `@event` binding, but it wouldn't work if the function was invoked in any other way, which we can't too easily know.

Another option is:

```ts
class MyElement extends LitElement {
  _onClick(event) {
    console.log('click', event.target);
  }
}
Object.assign(MyElement.prototype._onClick, {once: true});
```

cc @justinfagnani since we discussed this a bit in the meeting

Reply:
Hmm true, this works well for a Lit `@event` binding, but it wouldn't work if the function was invoked in any other way, which we can't too easily know.

Another option is:

```ts
class MyElement extends LitElement {
  _onClick(event) {
    console.log('click', event.target);
  }
}
Object.assign(MyElement.prototype._onClick, {once: true});
```

cc @justinfagnani since we discussed this a bit in the meeting

Reply:
Why not just:

```ts
MyElement.prototype._onClick.once = true;
```

Reply:
Yep that works too -- though I could see an argument that it could be nice for our idiomatic style to have a slightly closer match between the way you set event handler options in e.g. `addEventListener(..., {once: true, capture: true})` using an object expression.

Reply:
Switched to e.g. `@click=${{handleEvent: (e) => this._onClick(e), capture: true}}` as long as the handler is `private`, see comment from Kevin (and offline discussion). If not `private`, back to modifying the prototype method directly after the class declaration.